tests/unit/charms/tls_certificates_interface/v4/certificates.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import uuid
+from datetime import datetime, timedelta, timezone
+from typing import List, Optional
+
+from cryptography import x509
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import ec, rsa
+
+
+def csrs_match(csr_1: bytes, csr_2: bytes) -> bool:
```

Comment:
Many of the functions in this file seem to be unused. 

The others seem similar to the ones in `tls_certificates.py`. Perhaps this was done on purpose so that the tests don't rely on the implementations of those functions, but I wonder if if makes sense to just unit test those functions first, and then assume they are working for further tests so we aren't duplicating everything.

Reply:
I just removed those unused functions, they weren't added intentionally. Yes we don't want to rely on the src implementations for the "arrange" part of the tests but those were mistakes.